misc/xml-to-record-converter/src/main/java/io/ballerina/xmltorecordconverter/XMLToRecordConverter.java

```diff
@@ -239,46 +212,178 @@ private static void generateRecords(Element xmlElement, boolean isClosed,
-            recordToTypeDescNodes.put(xmlNodeName, recordTypeDescriptorNode);
         }
+        recordToTypeDescNodes.put(xmlNodeName, recordTypeDescriptorNode);
     }
 
-    private static RecordFieldNode getRecordField(Element xmlElementNode, List<String> existingFieldNames,
-                                                  Map<String, String> updatedFieldNames, boolean isOptionalField) {
+    private static List<Node> getRecordFieldsForXMLElement(Element xmlElement, boolean isClosed,
+                                                           Map<String, NonTerminalNode> recordToTypeDescNodes,
+                                                           Map<String, AnnotationNode> recordToAnnotationNodes,
+                                                           Map<String, Element> recordToElementNodes,
+                                                           List<DiagnosticMessage> diagnosticMessages) {
+        List<Node> recordFields = new ArrayList<>();
+
+        String xmlNodeName = xmlElement.getNodeName();
+        org.w3c.dom.NodeList xmlNodeList = xmlElement.getChildNodes();
+        for (int i = 0; i < xmlNodeList.getLength(); i++) {
+            org.w3c.dom.Node xmlNode = xmlNodeList.item(i);
+
+            if (xmlNode.getNodeType() == org.w3c.dom.Node.ELEMENT_NODE) {
+                Element xmlElementNode = (Element) xmlNode;
+                boolean isLeafXMLElementNode = isLeafXMLElementNode(xmlElementNode);
+                if (!isLeafXMLElementNode) {
+                    generateRecords(xmlElementNode, isClosed, recordToTypeDescNodes, recordToAnnotationNodes,
+                            recordToElementNodes, diagnosticMessages);
+                }
+                RecordFieldNode recordField = getRecordField(xmlElementNode, false);
+                if (recordFields.stream().anyMatch(recField -> ((RecordFieldNode) recField).fieldName().text()
+                        .equals(recordField.fieldName().text()))) {
+                    int indexOfRecordFieldNode = IntStream.range(0, recordFields.size())
+                            .filter(j -> ((RecordFieldNode) recordFields.get(j)).fieldName().text()
+                                    .equals(recordField.fieldName().text())).findFirst().orElse(-1);
+                    if (indexOfRecordFieldNode == -1) {
+                        recordFields.add(recordField);
+                    } else {
+                        RecordFieldNode existingRecordField =
+                                (RecordFieldNode) recordFields.remove(indexOfRecordFieldNode);
+                        RecordFieldNode updatedRecordField = mergeRecordFields(existingRecordField, recordField);
+                        recordFields.add(indexOfRecordFieldNode, updatedRecordField);
+                    }
+                } else {
+                    recordFields.add(recordField);
+                }
+            }
+        }
+        org.w3c.dom.NamedNodeMap xmlAttributesMap = xmlElement.getAttributes();
```

Comment:
Do we need to use fully qualified classnames for this class? 
```suggestion
        NamedNodeMap xmlAttributesMap = xmlElement.getAttributes();
```